src/state/slices/marketDataSlice/marketDataSlice.tsx

```diff
@@ -51,32 +52,42 @@ export const defaultMarketData: MarketData = {
   changePercent24Hr: 0,
 }
 
+type CryptoPriceHistoryPayload = { data: HistoryData[]; args: FindPriceHistoryByAssetIdArgs }
+
 export const marketData = createSlice({
   name: 'marketData',
   initialState,
   reducers: {
     clear: () => initialState,
-    setCryptoMarketData: (state, { payload }) => {
-      state.crypto.byId = { ...state.crypto.byId, ...payload } // upsert
-      const ids = Array.from(new Set([...state.crypto.ids, ...Object.keys(payload)]))
-      state.crypto.ids = ids // upsert unique
+    setCryptoMarketData: {
+      reducer: (state, { payload }: { payload: MarketDataById<AssetId> }) => {
+        state.crypto.byId = merge(state.crypto.byId, payload) // upsert
```

Comment:
lodash merge is deep, but i dont think the objects are partial so maybe we can get away with a shallow merge, e.g `state.crypto.byId = Object.assign(state.crypto.byId, payload)` ?